### PR TITLE
[SPARK-48449][SQL] Using `getPropertyInfo` instead of `reflection` to obtain `JAAS application name`

### PR DESCRIPTION
**Pull request by panbingkun**
### What changes were proposed in this pull request?
The pr aims to using `getPropertyInfo` instead of `reflection` to obtain `JAAS application name`, includes:
- MSSQLConnectionProvider
- PostgresConnectionProvider

### Why are the changes needed?
We should use the `public method` of the `Driver interface` and try to avoid using `private method` implemented by different `Drivers` to obtain `JAAS application name`

### Does this PR introduce _any_ user-facing change?
No.

### How was this patch tested?
- Existed UT
  MSSQLConnectionProviderSuite
  PostgresConnectionProviderSuite
- Pass GA.

### Was this patch authored or co-authored using generative AI tooling?
No.

### CODE REVIEW

**Comment by panbingkun** [comment]
cc @gaborgsomogyi

File: sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/jdbc/connection/MSSQLConnectionProvider.scala [hunk 1]
```diff
@@ -38,18 +38,16 @@ private[sql] class MSSQLConnectionProvider extends SecureConnectionProvider {
     val parseURL = try {
       // The default parser method signature is the following:
       // private Properties parseAndMergeProperties(String Url, Properties suppliedProperties)
-      val m = driver.getClass.getDeclaredMethod(parserMethod, classOf[String], classOf[Properties])
-      m.setAccessible(true)
```
Comment (dongjoon-hyun) [inline sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/jdbc/connection/MSSQLConnectionProvider.scala:42]: This looks irrelevant from the PR title. Why do we remove `setAccessible` here?

`parseAndMergeProperties` is `private`, isn't it? If this is not `private`, we need to change the comment at line 40 together.
